openpgp-sdkms/src/lib.rs

```diff
@@ -346,24 +368,60 @@ pub fn extract_cert(key_name: &str) -> Result<Cert> {
         }
     };
     if !metadata.contains_key(SDKMS_LABEL_PGP) {
-        return Err(Error::msg("malformed metadata in SDKMS".to_string()))
+        return Err(Error::msg("malformed metadata in SDKMS".to_string()));
     }
 
     let key_md: KeyMetadata = serde_json::from_str(&metadata[SDKMS_LABEL_PGP])?;
 
     Ok(Cert::from_str(&key_md.certificate)?)
 }
 
+pub fn extract_tsk_from_sdkms(key_name: &str) -> Result<Cert> {
+    // Extract all secrets as packets
+    let credentials = Credentials::new_from_env()?;
+    let http_client = credentials.http_client()?;
+
+    let mut packets = Vec::<Packet>::with_capacity(2);
+
+    // Primary key
+    let prim_sob = http_client
+        .export_sobject(&SobjectDescriptor::Name(key_name.to_string()))
```

Comment:
need some knowledge sharing: Looks like we don't have documentation for this method, do you know which part of code are we handling this request that we return different result for Name or UUID?

Reply:
Yes, this particular method is not documented on the client, but it's pretty self explanatory, and reading the documentation of [SdkmsClient](https://docs.rs/sdkms/0.2.1/sdkms/struct.SdkmsClient.html) ([source repo](https://github.com/fortanix/sdkms-client-rust)) can help.

If the Sobject was created with the `EXPORT` key operation, then the secret parts of the Sobjects are exportable. Upon calling `export_sobject`, you exfiltrate the full Sobject from DSM.

As for the [SobjectDescriptor](https://docs.rs/sdkms/0.2.1/sdkms/api_model/enum.SobjectDescriptor.html), it serves to identify the Sobject in question, you can identify it by name (as I'm doing here), by UUID, or as a Blob in the case of transient keys.

I can give you pointers for the backend code handling that request in another thread.

Reply:
Please let me know:) would like to have a look